core/arch/arm/kernel/thread_a32.S

```diff
@@ -748,6 +748,51 @@ END_FUNC thread_unwind_user_mode
 FUNC thread_vect_table , :
 UNWIND(	.fnstart)
 UNWIND(	.cantunwind)
+#ifdef CFG_CORE_WORKAROUND_SPECTRE_BP
+	/*
+	 * This depends on SP being 8 byte aligned, that is, the lowest
+	 * three bits in SP are zero.
+	 *
+	 * The idea is to form a specific bit pattern in the lowest three
+	 * bits of SP depending on which entry in the vector we enter via.
+	 * This is done by adding 1 to SP in each entry but the last.
+	 */
```

Comment:
I think this deserves a few words to explain that the BPIALL has to be executed before *any* branch instruction is executed, hence the slightly convoluted code. I know I could/should have made the same comment for the sm commit ;-)

Reply:
I'll add something.

Reply:
minor stuff: s/`doesn't`/`does not`/ , s/`it's`/`it is`/.

Sentence seems wrong to me: `It doesn't matter if it's a conditional or an unconditional branch speculation can still occur.`.

Reply:
Perhaps it's a missing comma. How about:
*It doesn't matter if it's a conditional or an unconditional branch, speculation can still occur.*